### ISSUE TITLE
ReferenceManyField - emptyText ignored

### ISSUE THREAD

**Issue by olliebennett** [marmelab/react-admin#6518]
There appears to be no way to display an empty state message when there are no results from a `<ReferenceManyField />`, despite docs suggesting the `emptyText` (common) field is supported.

**What you were expecting:**

When no results are returned for a referenced field, I expected my `<ReferenceManyField />` to display the text defined in its `emptyText` attribute.

**What happened instead:**

Nothing is displayed except the `<ReferenceManyField />`'s `label` attribute.

**Steps to reproduce:**

Using the code sandbox below, view the `show` screen of any Post without comments.

**Related code:**

CodeSandbox (forked from simple example) -> https://codesandbox.io/s/boring-swirles-zpxq5

Entirely unchanged except to add: 

```
emptyText='There are no comments for this post'
```

in `/posts/PostShow.tsx` line ~ `101`.

**Other information:**

The [ReferenceManyField docs](https://marmelab.com/react-admin/Fields.html#referencemanyfield) say that "<ReferenceManyField> also accepts the common field props." and `emptyText` is a common field prop.

**Environment**

* React-admin version: `3.17.2`
* Last version that did not exhibit the issue (if applicable): UNKNOWN
* React version: `17.0.2`
* Browser: Firefox
* Stack trace (in case of a JS error): `N/A`

**Comment by olliebennett** [comment]
I discovered the same is also true of `ArrayField`, and subsequently found discussion on https://github.com/marmelab/react-admin/issues/6179 covering the rationale.

TL;DR `emptyText` isn't supposed to work; it's a documentation issue.

The workaround is to use the child (eg. `Datagrid`'s) `empty` attribute;

```jsx
<ReferenceManyField ...>
  <Datagrid empty={<p>There are no comments for this post</p>}>
    ...
```

or, to style your text the same as a normal `emptyText`...

```jsx
import Typography from '@material-ui/core/Typography'

<ReferenceManyField ...>
  <Datagrid empty={<Typography component='span' variant='body2'>There are no comments for this post</Typography>}>
    ...
```

Feel free to close this issue, although hopefully it will help someone coming across this limitation in future.

**Comment by fzaninotto** [comment]
Yep, I'll close this issue as a duplicate of #6179. Thanks for the detailed report anyway!
